testkit/src/main/scala/org/bitcoins/testkit/chain/ChainUnitTest.scala

```diff
@@ -253,12 +261,24 @@ trait ChainUnitTest
                         hashBlockListener = None,
                         rawTxListener = None,
                         rawBlockListener = Some(handleRawBlock))
-    zmqSubscriber.start()
 
-    for {
-      _ <- AkkaUtil.nonBlockingSleep(1.second)
+    zmqSubscriber.start()
+    //generate a block and make sure we see it so we know the subscription is complete
+    val subscribedF = for {
+      _ <- chainHandlerF
+      addr <- bitcoind.getNewAddress
+      hash +: _ <- bitcoind.generateToAddress(1, addr)
       chainHandler <- chainHandlerF
+      //wait until we see the hash, to make sure the subscription is started
```

Comment:
This is the key thing to make sure we have zmq fully started and are receiving blocks